test/test_nv_exporter.py

```diff
@@ -0,0 +1,26 @@
+import unittest
+import shlex
+from helper import shell_process
+from helper import file_util
+
+
+"""Moneo Nvidia Exporter test"""
+
+class NvidiaExporterTestCase(unittest.TestCase):
+    """Nvidia Exporter unit test Class."""
+    def test_metrics(self):
+        """Check that required metrics are being tracked"""
+        metricData= file_util.load_data('data/nv_metric.txt')
+        metricData = metricData.split("\n")
+        #running nvidia exporter for 5 seconds with debug log level. It will out put the metrics monitored
+        cmd='timeout 5  python2 ../src/worker/exporters/nvidia_exporter.py --log-level DEBUG'
+        args = shlex.split(cmd)
```

Comment:
how do we handle timeout here?

Reply:
For this unit test I need the application to run for a designated amount of time since it runs in a loop outputting the metrics. The application would loop infinitely if launched without time out. So what this does is launch the exporter for 5 seconds. During those 5 seconds it will output metrics in a loop. I then check that the needed metrics are in that output.

Reply:
So no need for any additional handling.

Reply:
If there is an application time out the unit test would fail. I'd assume that would be desired results since you are checking to see exporter works correctly. Do you think it should be handled differently?

Reply:
Addressed